docs/current_docs/api/clients-sdk.mdx

```diff
@@ -48,8 +48,8 @@ dagger develop --sdk=typescript
 </TabItem>
 </Tabs>
 
-:::info
-At any point, running `dagger develop` regenerates the module's code based on dependencies and the current state of the module.
+:::warning
+Running `dagger develop` regenerates the module's code based on dependencies, the current state of the module, and the current Dagger version. This can result in unexpected results if there are significant changes between the previous and latest installed Dagger versions. Always refer to the [changelog](https://github.com/dagger/dagger/blob/main/CHANGELOG.md) for a complete list of changes (including breaking changes) in each Dagger release before running `dagger develop`.
```

Comment:
We could also document `--compat=skip` here? That skips the dagger version upgrade.